src/utils/listener/promptSecurity.ts

```diff
@@ -0,0 +1,123 @@
+import * as vscode from 'vscode';
+//For Prompt's server
+import express from 'express';
+import { Server } from "http";
+
+const MIN_CODE_LENGTH = 20;
+const MAX_CODE_LENGTH = 32 * 1024;
+
+export class PromptSecurity {
+	private hostname: string;
+	private port?: number;
+	private server?: Server;
+	private app?: express;
+	private context?:vscode.ExtensionContext;
+  
+	constructor(context:vscode.ExtensionContext,port:number) {
+	  this.hostname = "127.0.0.1";
+	  this.context = context;
+	  this.port = port;
+	  this.registerPromptListener();
+	}
+	getServer(){
+		return this.server;
+	}
+	deactivate(){
+		if (this.server){
+			this.server.close();
+		}
+	}
+	registerPromptListener(){
+		try {
```

Comment:
please make sure al l methods have try-catch

Reply:
Done, I also added try/catch to the PromptSecurity register in the extension.ts file